planner/core/testdata/integration_suite_out.json

```diff
@@ -5196,6 +5196,55 @@
       }
     ]
   },
+  {
+    "Name": "TestPushDownSelectionForMPP",
+    "Cases": [
+      {
+        "SQL": "desc format = 'brief' select /*+ hash_agg()*/ count(*) c, id from t group by id having id >c",
+        "Plan": [
+          "TableReader 6400.00 root  data:ExchangeSender",
+          "└─ExchangeSender 6400.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Selection 6400.00 mpp[tiflash]  gt(test.t.id, Column#5)",
+          "    └─Projection 8000.00 mpp[tiflash]  Column#5, test.t.id",
+          "      └─HashAgg 8000.00 mpp[tiflash]  group by:test.t.id, funcs:count(1)->Column#5, funcs:firstrow(test.t.id)->test.t.id",
+          "        └─ExchangeReceiver 10000.00 mpp[tiflash]  ",
+          "          └─ExchangeSender 10000.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.id, collate: binary]",
+          "            └─TableFullScan 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "desc format = 'brief' select * from t where id < 2",
+        "Plan": [
+          "TableReader 3323.33 root  data:ExchangeSender",
+          "└─ExchangeSender 3323.33 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Selection 3323.33 mpp[tiflash]  lt(test.t.id, 2)",
+          "    └─TableFullScan 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ]
+      }
+    ]
+  },
+  {
+    "Name": "TestPushDownSelectionForCop",
+    "Cases": [
+      {
+        "SQL": "desc format = 'brief' select /*+ hash_agg()*/ count(*) c, id from t group by id having id >c",
+        "Plan": [
+          "Selection 6400.00 root  gt(test.t.id, Column#5)",
```

Comment:
The selection is not push down to Cop, is it the expected result?

Reply:
Yes, it is expected. Because MPP is not allowed. HashAgg and Selection has to be executed on root task.

Reply:
I think this test case is meaningless, why not use a case that a selecton can not be pushed down to cop before this pr, and can be pushed down to cop after this pr?

Reply:
The next query is what you said. 
I will delete this one from TestPushDownSelectionForCop.

Reply:
But I think even without this pr, the next query will push selection to cop

Reply:
You are right, the case will pass even without this change. So it is indeed meaningless.

Reply:
I removed this piece of code.